tests/cpp/test_resize.cpp

```diff
@@ -3677,4 +3677,588 @@ TEST_F(ResizeTest, Chunk_Uneven) {
   EXPECT_EQ(out_tensors.back().numel(), 1);
 }
 
+// Schedule a slice with the loop domain derived from the producer
+// domain. See PR #2897.
+// Note that the IdModel-based indexing is automatically enabled as
+// there are tensors that have non-trivial loop domains as defined by
+// requiresIdModel in lower2device.cpp.
+TEST_F(ResizeTest, SliceScheduledLikeProducer) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  std::vector<int64_t> shape({100});
+
+  // concrete shapes to avoid dynamic Fusion
+  auto tv0 = makeConcreteTensor(shape);
+  fusion.addInput(tv0);
+
+  auto tv1 =
+      slice(tv0, {{IrBuilder::create<Val>(1L), IrBuilder::create<Val>(99)}});
+
+  auto tv2 = set(tv1);
+
+  fusion.addOutput(tv2);
+
+  tv1->setLoopDomain(tv1->getRootDomain());
+
+  auto tv2_loop_id = tv0->getLoopDomain().at(0)->cloneWithoutRFactor();
+
+  IrBuilder::create<Resize>(
+      tv2->getLogicalDomain().at(0),
+      tv2_loop_id,
+      IrBuilder::create<Val>(-1, DataType::Index),
+      IrBuilder::create<Val>(-1, DataType::Index));
+
+  tv2->setLoopDomain({tv2_loop_id});
+
+  for (auto tv : {tv1, tv2}) {
+    tv->split(0, 32);
+    // RAW sync analysis not updated yet
+#if 0
+    tv->axis(0)->parallelize(ParallelType::BIDx);
+    tv->axis(1)->parallelize(ParallelType::TIDx);
+#endif
+  }
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+
+  auto t0 = at::randn(shape, options);
+  std::vector<c10::IValue> aten_inputs({t0});
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, aten_inputs);
+  auto cg_outputs = fe.runFusion(aten_inputs);
+
+  auto ref = t0.index({at::indexing::Slice(1, shape[0] - 1)});
+
+  NVF_CHECK(ref.equal(cg_outputs[0]));
+}
+
+TEST_F(ResizeTest, PadScheduledLikeConsumer) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  std::vector<int64_t> shape({100});
+
+  // concrete shapes to avoid dynamic Fusion
+  auto tv0 = makeConcreteTensor(shape);
+  fusion.addInput(tv0);
+
+  auto tv1 = add(tv0, IrBuilder::create<Val>(1));
+  auto tv2 = pad(tv1, {IrBuilder::create<Val>(1), IrBuilder::create<Val>(1)});
+
+  auto tv3 = add(tv2, IrBuilder::create<Val>(1));
+  fusion.addOutput(tv3);
+
+  auto tv1_padded = IterDomain::resize(
+      tv1->getLoopDomain().at(0),
+      IrBuilder::create<Val>(1, DataType::Index),
+      IrBuilder::create<Val>(1, DataType::Index));
+  tv1->setLoopDomain({tv1_padded});
```

Comment:
I sorta see the need to create tv1_padded, Just for clarification, is there a requirement about the dependency of LoopDomain? i.e. can we use the domain of its consumer as the loop domain of the producer?

i.e. `tv1->setLoopDomain(tv2->getLogicalDomain())`

Reply:
As discussed offline, it's because each tensor needs its own domains and exprs, at this moment.